### DISCUSSION TITLE
Add support to Metadata for local files and resources in Media Element

### DISCUSSION THREAD

**Discussion by ne0rrmatrix** [CommunityToolkit/Maui#1989] [New Feature Discussions]
I am hoping to get support from the community with votes so I can start the process of adding support for local file and Media Resources to the list of options for metadata sources for media element. ATM it only supports url's. This was planned to make supporting the initial release easier. Now that it is out I want to extend the metadata source to include the full range of urls, local files, and package resources as options for metadata. ATM poster is the only field it applies to and it will be easy to add this support.

**Comment by JohnStrudwick** [comment]
I would support this new feature - the internet is not always available or fast, so a URL is not always the best. 

Certainly for an audio player using MP3 files, you can read the artwork directly from the MP3 file itself. I am using this so that a user can see this by pressing a button. It would be nice to see this in the metadata. Ideally reading from a stream or a local file.

For reference, I use the TagLib-Sharp library (from nuget) to extract the artwork as the source for an Image with this code:
XAML:
<Image
	x:Name="Mp3Image"
/>
Code behind:

TagLib.File tagFile = TagLib.File.Create(FileName);
if(tagFile.Tag.Pictures.Length > 0)
{
	var Source = ImageSource.FromStream(() => new MemoryStream(tagFile.Tag.Pictures[0].Data.Data));
	Mp3Image.Source = Source;
	Mp3Image.IsVisible = true;
}
